packages/edge/src/edge-headers.ts

```diff
@@ -67,6 +78,22 @@ export function ipAddress(request: Request): string | undefined {
   return getHeader(request, IP_HEADER_NAME);
 }
 
+/**
+ * Extracts the Vercel Edge Network region name from the request ID.
+ *
+ * @param requestId The request ID (`x-vercel-id`).
+ * @returns The first region received the client request.
+ */
+function getRegionFromRequestId(requestId?: string): string | undefined {
+  if (!requestId) {
+    return 'dev1';
+  }
+
+  // The request ID is in the format of `region::id` or `region1:region2:...::id`.
+  const match = requestId.match(/^(?<region>[a-z0-9]+):/);
+  return match?.groups?.region;
```

Comment:
can this be `requestId.split(':')[0]`? what is the benefit of regex here?

Reply:
I thought this regex approach is more robust but there's not much difference as you may think. Will rewrite this with `split` 👍